src/System.Memory/tests/Performance/Perf.ReadOnlySequence.Slice.cs

```diff
@@ -2,765 +2,338 @@
-            memory = memory.Slice(bufOffset, bufSize - 2 * bufOffset);
-            var buffer = new ReadOnlySequence<char>(memory);
-            long offset = buffer.Length / 10;
+            var result = new List<SequencePosition>(count);
 
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            List<long> positions = CookPositionsLong(buffer, count);
+            foreach (long startOffset in positions)
             {
-                int localInt = 0;
-                using (iteration.StartMeasurement())
-                {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ReadOnlySequence<char> temp = buffer.Slice(offset);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
+                SequencePosition start = buffer.GetPosition(startOffset);
+                result.Add(start);
             }
-        }
 
-        [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void String_LongLong(int bufSize, int bufOffset)
-        {
-            ReadOnlyMemory<char> memory = new string('a', bufSize).AsMemory();
-            memory = memory.Slice(bufOffset, bufSize - 2 * bufOffset);
-            var buffer = new ReadOnlySequence<char>(memory);
-            long offset = buffer.Length / 20;
-            long sliceLen = buffer.Length - 2 * offset;
-
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
-            {
-                int localInt = 0;
-                using (iteration.StartMeasurement())
-                {
-                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
-                    {
-                        ReadOnlySequence<char> temp = buffer.Slice(offset, sliceLen);
-                        localInt ^= temp.Start.GetInteger();
-                    }
-                }
-                _volatileInt = localInt;
-            }
+            return result;
         }
 
-        [Benchmark(InnerIterationCount = InnerCount)]
-        [InlineData(10_000, 100)]
-        private static void String_LongPos(int bufSize, int bufOffset)
+        internal static List<(long, long)> CookPositionsLongLong(in ReadOnlySequence<T> buffer, int count)
         {
-            ReadOnlyMemory<char> memory = new string('a', bufSize).AsMemory();
-            memory = memory.Slice(bufOffset, bufSize - 2 * bufOffset);
-            var buffer = new ReadOnlySequence<char>(memory);
-            long offset = buffer.Length / 20;
-            SequencePosition end = buffer.GetPosition(0, buffer.End);
+            var result = new List<(long, long)>(count);
 
-            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            long length = buffer.Length;
+            long offset = length / (2 * count);
```

Comment:
Can you please add comments explaining why offset, length, etc. is set to these values?